core/src/main/java/com/linecorp/armeria/client/endpoint/OrElseEndpointGroup.java

```diff
@@ -18,25 +18,27 @@
 
 import static java.util.Objects.requireNonNull;
 
-import java.util.List;
+import java.util.Collection;
 
 import com.linecorp.armeria.client.Endpoint;
 
-final class OrElseEndpointGroup implements EndpointGroup {
+final class OrElseEndpointGroup extends EndpointGroup {
     private final EndpointGroup first;
     private final EndpointGroup second;
 
     OrElseEndpointGroup(EndpointGroup first, EndpointGroup second) {
         this.first = requireNonNull(first, "first");
         this.second = requireNonNull(second, "second");
+        updateEndpoints(null);
```

Comment:
I see. Sounds good to me. @imasahiro How about making `EndpointGroup` an interface again (we keep the notification operations this time) so that `EndpointGroup` implementations have more freedom?